src/main/java/com/example/demo/bo/UserAdvanceDetailsBO.java

```diff
@@ -3,13 +3,17 @@
 import java.util.List;
 import java.util.Map;
 
+import org.springframework.stereotype.Component;
+
+@Component
 public class UserAdvanceDetailsBO extends UserBasicDetailsBO {
 
 	private Map<String, UserAcademicDetailsBO> userAcademicDetails; // Map of Academic id to Academic Details
 	private List<LeaveDetailsBO> userApplicableLeaves;
 	private List<UserManagerDetailsBO> userManagerDetails;
 	private Map<String, PayrollDetailsBO> userPayrollDetails; // Map of Academic Id to Payroll details
```

Comment:
@ssjoshiy2k  change userPayrolDetails map to map<String, List<PayrollDetailsBO>>